libraries/protocol/steem_operations.cpp

```diff
@@ -107,37 +107,40 @@ namespace golos { namespace protocol {
         };
 
         void comment_payout_beneficiaries::validate() const {
-            uint32_t sum = 0;
-
-            FC_ASSERT(beneficiaries.size(), "Must specify at least one beneficiary");
-            FC_ASSERT(beneficiaries.size() < 128,
-                      "Cannot specify more than 127 beneficiaries."); // Require size serializtion fits in one byte.
-
-            validate_account_name(beneficiaries[0].account);
-            FC_ASSERT(beneficiaries[0].weight <= STEEMIT_100_PERCENT,
-                      "Cannot allocate more than 100% of rewards to one account");
-            sum += beneficiaries[0].weight;
-            FC_ASSERT(sum <= STEEMIT_100_PERCENT,
-                      "Cannot allocate more than 100% of rewards to a comment"); // Have to check incrementally to avoid overflow
-
-            for (size_t i = 1; i < beneficiaries.size(); i++) {
-                validate_account_name( beneficiaries[i].account);
-                FC_ASSERT(beneficiaries[i].weight <= STEEMIT_100_PERCENT,
-                          "Cannot allocate more than 100% of rewards to one account");
-                sum += beneficiaries[i].weight;
-                FC_ASSERT(sum <= STEEMIT_100_PERCENT,
-                          "Cannot allocate more than 100% of rewards to a comment"); // Have to check incrementally to avoid overflow
-                FC_ASSERT(beneficiaries[i - 1] < beneficiaries[i],
-                          "Benficiaries must be specified in sorted order (account ascending)");
-            }
+            fc::safe<uint32_t> sum = 0;
+
+            GOLOS_CHECK_PARAM(beneficiaries, {
+                GOLOS_CHECK_VALUE(beneficiaries.size(), "Must specify at least one beneficiary");
+                GOLOS_CHECK_VALUE(beneficiaries.size() < 128,
+                          "Cannot specify more than 127 beneficiaries."); // Require size serializtion fits in one byte.
+
+                try {
+                    for (auto beneficiar: beneficiaries) {
+                        validate_account_name(beneficiaries[0].account);
+                        GOLOS_CHECK_VALUE(beneficiar.weight <= STEEMIT_100_PERCENT,
+                                "Cannot allocate more than 100% of rewards to one account");
+                        sum += beneficiar.weight;
+                    }
+
+                    GOLOS_CHECK_VALUE(sum <= STEEMIT_100_PERCENT,
+                            "Cannot allocate more than 100% of rewards to a comment");
+                } catch(const fc::overflow_exception& e) {
```

Comment:
overflow is impossible here even without `safe`, so this try/catch is not required

Reply:
You are right, overflow in this place impossible. 1000*127 = 1'270'000 (<2*10^9). But this error can occur after some modifications in future (say, increase limit on beneficiaries count). So, it's need to use `fc::safe`. In this case, client gets internal error.

Reply:
`FC_ASSERT(sum <= STEEMIT_100_PERCENT` was placed **inside** loop specially for this case